go/auth/token.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2015 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+//
+// Code used to support authentication tokens for arbitrary purposes.
+//
+package auth
+
+import (
+	"bytes"
+	"encoding/json"
+	"math"
+	"time"
+
+	libkb "github.com/keybase/client/go/libkb"
+	keybase1 "github.com/keybase/client/go/protocol"
+)
+
+const CurrentTokenVersion = 1
+
+type TokenKey struct {
+	UID      keybase1.UID             `json:"uid"`
+	Username libkb.NormalizedUsername `json:"username"`
```

Comment:
I wonder, will this work over JSON?  We might lack the necessary hooks to Decode and Encode this type. I'm not precisely sure though.  We'll see in integration testing.